### PR TITLE
[5|Resource Page (Item) and Note Taking Feature] The tooltip 

### PR DESCRIPTION
**Pull request by ViniOkamoto**
PR for #14 

- Added component for a tooltip when the user selects the content inside of our resource page.
- Added functionality to the copy button

image

image

Output: "This is the resource"

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,6 +55,7 @@
     "nextjs-toploader": "^1.6.4",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-tooltip": "^5.28.0",
```
Comment (Simpcyclassy) [inline package.json:58]: Looks like this i not used anywhere.

File: src/components/ResourceContentTip/index.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        const tooltipWidth = tooltipRef.current
+          ? tooltipRef.current.offsetWidth
+          : 0
+
+        // Calculate the left position with boundary checks
+        console.log('range', range)
+        console.log('window.scrollX', window.scrollX)
+        console.log('range.width', range.width)
+        console.log('tooltipWidth', tooltipWidth)
+        console.log('viewportWidth', window.innerWidth)
```
Comment (Simpcyclassy) [inline src/components/ResourceContentTip/index.tsx:42]: Remove unnecessary console logs

File: src/components/ResourceContentTip/index.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        let leftPos = range.left + window.scrollX + range.width - tooltipWidth
+
+        // Ensure the tooltip stays within the viewport
+        const viewportWidth = window.innerWidth
+        console.log(viewportWidth)
+        console.log(leftPos + tooltipWidth > viewportWidth)
+        console.log(leftPos + tooltipWidth)
```
Comment (Simpcyclassy) [inline src/components/ResourceContentTip/index.tsx:49]: These as well

**Comment by ViniOkamoto** [comment]
@Simpcyclassy added the changes as requested and improved comment for further devs

**Comment by Simpcyclassy** [comment]
> @Simpcyclassy added the changes as requested and improved comment for further devs

@ViniOkamoto Can I see you to better understand what you mean?

**Review by Simpcyclassy — APPROVED** [review]
LGTM

**Pull request merged**